crates/bevy_winit/src/lib.rs

```diff
@@ -54,8 +55,8 @@ impl Plugin for WinitPlugin {
                 CoreStage::PostUpdate,
                 SystemSet::new()
                     .label(ModifiesWindows)
-                    .with_system(changed_window)
-                    .with_system(despawn_window),
+                    .with_system(changed_window.before(exit_on_all_closed))
```

Comment:
what's the logic behind the ordering of these three systems?

Reply:
Rationale has been added in https://github.com/bevyengine/bevy/pull/7383/commits/1f72ad51b9d30cae31f105b22d4fbe5e017cbfe8

Reply:
I'm not sure the ordering here matters at all. despawn_window and changed_window are conflicting on WinitWindows, but they will always act on different windows. despawn_window is using RemovedComponents<Window> while changed_window works with Changed<Window>. These will always be mutually exclusive.

For changed_window and exit_on_all_closed is a similar situation. exit_on_all_closed is just checking if the Query<&Window> query is empty. Which should be mutually exclusive with Changed<Window>.

Reply:
> despawn_window is using RemovedComponents<Window> while changed_window works with Changed<Window>. These will always be mutually exclusive.

This is not strictly true: you can remove and then readd the component in the same frame.

Reply:
> This is not strictly true: you can remove and then readd the component in the same frame.

That's a good point. What would happen in that case then? Would the window be removed in winit, even though there's an entity with Window on it?

Reply:
Yes. Let me add a fix for that...

Reply:
Alright, fixed. I still think that we should be deterministic here, but you've convinced me that there's no clear argument for one before the other.